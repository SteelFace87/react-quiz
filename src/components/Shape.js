import React from 'react';
import PropTypes from 'prop-types';

export default function Shape({ text, color, backgroundColor }){

  return (
  <>
  <p style={{ fontColor: color }}>text:{text}</p>
  <div style={{ width:'100px', height:'100px', backgroundColor:backgroundColor }}></div>
  </>
  );

}

Shape.propTypes = {
  text:PropTypes.string.isRequired,
  color:PropTypes.string.isRequired,
  backgroundColor:PropTypes.string.isRequired

}
;
