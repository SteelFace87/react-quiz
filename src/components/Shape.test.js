import React from 'react';
import { shallow } from 'enzyme';
import Shape from './Shape';

describe('Shape', ()=>{
  it('renders stuff', ()=>{
    const wrapper =  shallow(<Shape text='test' color='green' backgroundColor='blue'/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
