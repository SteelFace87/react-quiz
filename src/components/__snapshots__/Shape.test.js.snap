// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape renders stuff 1`] = `
<Fragment>
  <p
    style={
      Object {
        "color": "green",
      }
    }
  >
    text:
    test
  </p>
  <div
    style={
      Object {
        "backgroundColor": "blue",
        "height": "100px",
        "width": "100px",
      }
    }
  />
</Fragment>
`;
